src/components/header/EmptyThreadIndicator.js

```diff
@@ -126,4 +75,72 @@ class EmptyThreadIndicator extends PureComponent<Props> {
   }
 }
 
+/**
+ * Define this outside of the class so that it's easily testable. The internals
+ * are a little complicated with the math, but the test file should have some
+ * pretty clear explanations of the requirements:
+ * src/test/components/EmptyThreadIndicator.test.js
+ */
+export function getIndicatorPositions(
+  props: Props
+): {|
+  startup: SyntheticCssDeclarations | null,
+  shutdown: SyntheticCssDeclarations | null,
+  emptyBufferStart: SyntheticCssDeclarations | null,
+|} {
+  const {
+    rangeStart,
+    rangeEnd,
+    width,
+    unfilteredSamplesRange,
+    interval,
+    thread: { processShutdownTime, registerTime, unregisterTime },
+  } = props;
+  const rangeLength = rangeEnd - rangeStart;
```

Comment:
Maybe this is not a big deal, but... is `rangeEnd` included in the range, or not? If it's included, we need a `+ interval`.

Reply:
I tracked it down, and it includes the interval already:

https://github.com/devtools-html/perf.html/blob/master/src/profile-logic/profile-data.js#L139-L150